src/live_restore/live_restore_fs.c

```diff
@@ -763,22 +761,19 @@ __live_restore_fh_read(
         WT_RET(__live_restore_fh_read_destination(session, lr_fh->destination, offset, len, buf));
 
     __wt_readlock(session, &lr_fh->lock);
-    /*
-     * The partial read length variables need to be initialized inside the else case to avoid clang
-     * sanitizer complaining about dead stores. However if we use a switch case here _and_
-     * initialize the variables inside the PARTIAL case then gcc complains about the switch jumping
-     * over the variable declaration. Thus we use if/else and declare inside to keep both happy.
-     */
     wt_off_t hole_begin_off;
-    WT_LIVE_RESTORE_SERVICE_STATE read_state =
-      __live_restore_can_service_read(lr_fh, session, offset, len, &hole_begin_off);
-    if (read_state == FULL)
+    if (__live_restore_can_service_read(session, lr_fh, offset, len, &hole_begin_off)) {
+        /* Read the full read from the destination. */
         WT_ERR(
           __live_restore_fh_read_destination(session, lr_fh->destination, offset, len, read_data));
-    else if (read_state == PARTIAL) {
+    } else {
+        /* Otherwise from the source. */
+        WT_ERR(__live_restore_fh_read_source(session, lr_fh->source, offset, len, read_data));
+
+#ifdef HAVE_DIAGNOSTIC
         /*
-         * If a portion of the read region is serviceable, combine a read from the source and
-         * destination.
+         * If a portion of the read region is serviceable then it is a partial read, add a check
+         * here to ensure partial reads always identical to the reads from the source.
```

Comment:
```suggestion
         * If we found a read can be partially serviced it's because the page has been partially migrated by a background thread. This means the data in the source and destination should be identical, and we verify this as a safety check.
```